src/index.js

```diff
@@ -1,40 +1,87 @@
 const list = document.getElementById('list');
-const position = 'beforeend';
-
-const toDoList = [
-  {
-    index: 0,
-    description: 'Reading books',
-    completed: true,
-  },
-  {
-    index: 1,
-    description: 'Going for a walk',
-    completed: true,
-  },
-  {
-    index: 2,
-    description: 'Meet families',
-    completed: true,
-  },
-  {
-    index: 3,
-    description: 'play football',
-    completed: true,
-  },
-];
-
-const todo = () => {
-  for (let i = 0; i < toDoList.length; i += 1) {
-    const item = `<li class="item">
+const input = document.getElementById('input');
+const addBtn = document.querySelector('.add');
+
+let LIST;
+let id;
+
+const check = 'fa-square-check';
+const uncheck = 'fa-square';
+const line = 'line-through';
+// local storage
+
+function addToDo(todo, id, done, trash) {
+  if (trash) { return; }
+  const Done = done ? check : uncheck;
+  const Line = done ? line : '';
+  const position = 'beforeend';
+
+  const item = `<li class="item">
               <div class="left">
-              <i class="fa-regular fa-square"></i>
-              <p class="text">${toDoList[i].description}</p>
+              <i class="fa-regular ${Done}" job="complete" id=${id}></i>
+              <p class="text ${Line}">${todo}</p>
               </div>
-              <i class="fa-solid fa-trash" class="complete"></i>
+              <i class="fa-solid fa-trash" job="delete" id=${id}></i>
             </li>`;
-    list.insertAdjacentHTML(position, item);
-  }
+
+  list.insertAdjacentHTML(position, item);
+}
+
+const loadList = (array) => {
+  array.forEach((item) => {
+    addToDo(item.todo, item.id, item.done, item.trash);
+  });
 };
+const myData = localStorage.getItem('TODO');
+if (myData) {
+  LIST = JSON.parse(myData);
+  id = LIST.length;
+  loadList(LIST);
+} else {
+  LIST = [];
+  id = 0;
+}
+
+addBtn.addEventListener('click', () => {
+  const todo = input.value;
+
+  if (todo) {
+    addToDo(todo, id, false, false);
+    LIST.push({
+      todo,
+      id,
+      done: false,
+      trash: false,
+    });
+
+    id += 1;
+    localStorage.setItem('TODO', JSON.stringify(LIST));
+  }
+
+  input.value = '';
+});
+
+function completeToDo(element) {
+  element.classList.toggle(check);
+  element.classList.toggle(uncheck);
+  element.parentNode.querySelector('.text').classList.toggle(line);
+  LIST[element.id].done = !LIST[element.id].done;
+}
+
+function removeToDo(element) {
+  element.parentNode.parentNode.removeChild(element.parentNode);
+  LIST[element.id].trash = true;
+}
+
+list.addEventListener('click', (event) => {
+  const element = event.target;
+  const elementJob = element.attributes.job.value;
+  if (elementJob === 'complete') {
+    completeToDo(element);
+  } else if (elementJob === 'delete') {
+    removeToDo(element);
+  }
+  localStorage.setItem('TODO', JSON.stringify(LIST));
+});
```

Comment:
- [ ] Please make sure to place all code for this project's functionalities in a new file as requested in the project requirements.

![image](https://user-images.githubusercontent.com/83803414/222383493-e4eb51ae-a7c4-497f-b8c3-dac50e67d2a5.png)

Reply:
- Well resolved 👍🏽